apollo-router/src/plugins/cache/entity.rs

```diff
@@ -40,15 +41,44 @@ register_plugin!("apollo", "experimental_entity_cache", EntityCache);
 
 struct EntityCache {
     storage: RedisCacheStorage,
+    subgraphs: Arc<HashMap<String, Subgraph>>,
+    enabled: Option<bool>,
 }
 
 /// Configuration for entity caching
 #[derive(Clone, Debug, JsonSchema, Deserialize)]
 #[serde(rename_all = "snake_case", deny_unknown_fields)]
 struct Config {
     redis: RedisCache,
+    /// activates caching for all subgraphs, unless overriden in subgraph specific configuration
+    #[serde(default)]
+    enabled: Option<bool>,
+    /// Per subgraph configuration
+    #[serde(default)]
+    subgraphs: HashMap<String, Subgraph>,
 }
 
+/// Per subgraph configuration for entity caching
+#[derive(Clone, Debug, JsonSchema, Deserialize)]
+#[serde(rename_all = "snake_case", deny_unknown_fields)]
+struct Subgraph {
```

Comment:
Nit could we get away with serde default at the struct level and implement `Default` for subgraph?

A long TTL and enabled true?

Reply:
This is one of those cases where the subgraph config has to override the global config, but only for the fields specified in the subgraph, so I can't set default values